### PR TITLE
Refactor/middleware

### PR DESCRIPTION
**Pull request by steinalex**
Middleware has been refactored to be part of each feature folder. Socket service is now passed down as a function to each middleware

### CODE REVIEW

File: src/features/chart/redux/middleware.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { UPDATE_CHART_RANGE } from "../../../store/constants";
+
+export const chartMiddleware = socketService => store => next => action => {
```
Comment (Sharakai) [inline src/features/chart/redux/middleware.js]: FWIW:
I would change this method signature slightly:
```
export const chartMiddleware = ({ socketService }) => store => next => action => {
   ...
```
It'll allow you to extend that initial "dependencies" parameter, if you need to pass in additional services.

File: src/store/middleware.js [hunk 2]
```diff
@@ -14,61 +9,27 @@ import { updateLatestNewsAction } from "../features/latest-news";
...
-      .get()
-      .emit(
-        "timeRange",
-        store.getState().stockData.selectedStock.symbol,
-        action.payload
-      );
-  } else if (action.type === UPDATE_SEARCH_QUERY) {
-    socketService.get().emit("searchQuery", action.payload);
+    dataTofetch.forEach(([name, action]) => {
+      socket.on(name, payload => {
```
Comment (Sharakai) [inline src/store/middleware.js]: Why not have each of your other middleware listening to the `BOOTSTRAP` action and do this on their own?

File: src/features/chart/redux/middleware.js (hunk 1 above)
Comment (steinalex) [reply src/features/chart/redux/middleware.js]: Great point - will change

File: src/store/middleware.js (hunk 2 above)
Comment (steinalex) [reply src/store/middleware.js]: I'd prefer to have a centralized place to start every component

Comment (Sharakai) [reply src/store/middleware.js]: Doesn't this mean that this file could become out of hand as features are added?
Why have multiple middleware at all?

Comment (steinalex) [reply src/store/middleware.js]: Very good point - will raise in a new PR

**Pull request merged**